server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java

```diff
@@ -158,8 +158,29 @@
          * @param subscribedColumns are the columns subscribed for this view
          * @return a MessageView filtered by the subscription properties that can be sent to that subscriber
          */
-        MessageView getSubView(Options options, boolean isInitialSnapshot, @Nullable RowSet viewport,
+        MessageView getSubView(BarrageSubscriptionOptions options, boolean isInitialSnapshot, @Nullable RowSet viewport,
                 @Nullable RowSet keyspaceViewport, BitSet subscribedColumns);
+
+        /**
+         * Obtain a Full-Subscription View of this StreamGenerator that can be sent to a single subscriber.
```

Comment:
Both of these methods need a bit of javadoc and parameter cleanup to be snapshot not subscription.